test/TestSemanticAnalyzer.hs

```diff
@@ -0,0 +1,94 @@
+module TestSemanticAnalyzer  ( module TestSemanticAnalyzer
+        ) where
+
+    import           Test.Hspec
+    import           Semantic.SemanticAnalyzer
+    import qualified Semantic.SymbolTable as ST
+    import qualified Data.Map as Map
+    import           AST
+    
+    defaultSymbolTable :: ST.SymbolTable
+    defaultSymbolTable = Map.empty
+
+    -- Test for analyzeTypeIndentifier
+    testTypeIdentifier :: Spec
+    testTypeIdentifier = describe "analyzeTypeIndentifier" $ do
+        it "adds a new symbol to the symbol table" $ do
+            let typedId = TypedIdentifier "x" TInt
+            let expectedMap = Map.fromList [("x", ST.Symbol "x" ST.INTEGER (ST.IntSymbolValue 0))]
+            analyzeTypeIndentifier typedId defaultSymbolTable `shouldBe` (typedId, expectedMap)
+
+    -- Test for analyzeLetStatement
+    testLetStatement :: Spec
+    testLetStatement = describe "analyzeLetStatement" $ do
+        it "adds a let statement to the symbol table" $ do
+            let typedId = TypedIdentifier "y" TInt
+            let letStatement = LetStatement typedId (ValueExpr (VLiteral (IntLiteral 10)))
+            let expectedMap = Map.fromList [("y", ST.Symbol "y" ST.INTEGER (ST.IntSymbolValue 0))]
+            analyzeLetStatment letStatement defaultSymbolTable `shouldBe` (letStatement, expectedMap)
```

Comment:
I see you define in line 26 a let statement with the value of 10, but in the line 27 you wrote 0. Is this the missing part of the analyzer you mentioned us in the group?

Reply:
Got it!